test/test_dir.rs

```diff
@@ -0,0 +1,46 @@
+extern crate nix;
+extern crate tempdir;
+
+use nix::dir::{Dir, Type};
+use nix::fcntl::OFlag;
+use nix::sys::stat::Mode;
+use std::fs::File;
+use self::tempdir::TempDir;
+
+#[test]
+fn read() {
+    let tmp = TempDir::new("nix-dir-read").unwrap();
```

Comment:
Can't we have anonymous folders? I'd prefer we use that style rather than naming everything.

Reply:
likewise